music-journal/pages/api/list-memories.js

```diff
@@ -1,30 +1,39 @@
-export default async function handler(req, res) {
-    const firebaseAdmin = require('firebase-admin');
-    const firebase = require('firebase');
-    const serviceAccount = require('../../service-account.json');
-    
-    if (!firebaseAdmin.apps.length) {
-        firebaseAdmin.initializeApp({
-            credential: firebaseAdmin.credential.cert(serviceAccount),
-            databaseURL: process.env.FIREBASE_DATABASE_URL,
-        });
-    }
+import { auth } from 'firebase'
+import Cors from 'cors';
+import { database } from '../../utils/initFirebaseAdmin'
+import initMiddleware from '../../utils/initMiddleware'
 
-    const user = await firebase.auth().currentUser;
-    if (!user) {
-        res.status(403).json({
-            authenticated: false
-        });
-    }
+const cors = initMiddleware(
+    Cors({
+        methods: ['GET', 'POST', 'OPTIONS', 'HEAD'],
+    })
+)
+
+async function handler(req, res) {
+    await cors(req, res)
+
+    try {
+        const user = auth().currentUser;
 
-    const ref = await firebaseAdmin.database().ref(`/memory/${user.uid}`);
-    await ref.orderByValue().once("value", snapshot => {
-        let memories = [];
-        snapshot.forEach(data => {
-            memories.push(data.val());
-        });
-        res.send(
-            memories
-        );
-    });
+        if (!user) {
+            console.log('list-memories user')
+            res.status(403).json({
+                authenticated: false
+            });
+        }
+
+        let memories = []
+        
+        const memoriesRef = await database.ref(`/memory/${user.uid}`);
+        
+        await memoriesRef.orderByValue().once("value", snapshot => {
+            snapshot.forEach(data => memories.push(data.val()))
```

Comment:
nice!

Reply:
I still wanna find a way for it to just return the list rather than just nest it in the chained functions but I think this is good enough